### ISSUE TITLE
Allow form-control-sm/form-control-lg to be passed to Select2 form elements

### ISSUE THREAD

**Issue by magicsunday** [fisharebest/webtrees#2056]
- Select2 form elments contains duplicate CSS class "form-control"
  - https://dev.webtrees.net/demo-dev/index.php?route=pedigree&xref=i1&ged=demo
- The class does not respect extra CSS classes passed via "class" in the attributes list of `FunctionsEdit::formControlIndividual`, thus its not possible to change the layout of the form elements.
- If adding for instance "form-control-sm" to `Select2::commonConfig` to get smaller form elements, the element itself overwrites the height again (in styles.css), also the image size does not match and should be adjusted imho.
- Why is the width hardcoded to 90%?

**Comment by magicsunday** [comment]
Possible alternatives for select2:

- https://joshuajohnson.co.uk/Choices/
- https://selectize.github.io/selectize.js/
  - https://github.com/const-se/selectize-bootstrap4-theme

Or trying one of the bootstrap4-select2 themes.

**Comment by fisharebest** [comment]
> Why is the width hardcoded to 90%?

Not sure.  I think it was a workaround for input-field addons in an earlier version of bootstrap.  I've taken it out.

**Comment by fisharebest** [comment]
I'd like to remove this class, and replace it with views.  (I already did this for the `Datatables` class).

Select2 is awful for accessibility.

Last time I looked, selectize appeared to be abandoned - but I see there is some recent activity.

twitter typeahead/bloodhound is abandoned.

chosenjs is great for accessibility - but has no native support for ajax data sources.  See issue 79 on its github page.
